examples/joke_bot/src/main.rs

```diff
@@ -0,0 +1,268 @@
+    },
+    presence::PresenceState,
+    serde::Raw,
+    RoomId, UserId,
+};
+use serde_json::Value as JsonValue;
+use tokio::fs;
+use tokio_stream::StreamExt as _;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    if let Err(e) = run().await {
+        println!("{}", e);
+        exit(1)
+    }
+    Ok(())
+}
+
+type HttpClient = client::http_client::HyperNativeTls;
+type MatrixClient = client::Client<http_client::HyperNativeTls>;
+
+async fn run() -> Result<(), Box<dyn Error>> {
+    let config =
+        read_config().await.map_err(|e| format!("configuration in ./config is invalid: {}", e))?;
+    let http_client =
+        hyper::Client::builder().build::<_, hyper::Body>(hyper_tls::HttpsConnector::new());
+    let matrix_client = if let Some(state) = read_state().await.ok().flatten() {
+        MatrixClient::with_http_client(
+            http_client.clone(),
+            config.homeserver.to_owned(),
+            Some(state.access_token),
+        )
+    } else if let Some(password) = &config.password {
+        let client =
+            MatrixClient::with_http_client(http_client.clone(), config.homeserver.to_owned(), None);
+        match client.log_in(config.username.as_ref(), password, None, None).await {
+            Ok(_) => {
+                if let Err(err) = write_state(&State {
+                    access_token: client.access_token().expect("Matrix access token is missing"),
+                })
+                .await
+                {
+                    println!("Failed to persist access token to disk. Re-authentication will be required on the next startup: {}", err)
+                }
+                client
+            }
+            Err(e) => {
+                let reason = match e {
+                    client::Error::AuthenticationRequired => {
+                        "invalid credentials specified".to_owned()
+                    }
+                    client::Error::Response(response_err) => {
+                        format!("failed to get a response from the server: {}", response_err)
+                    }
+                    client::Error::FromHttpResponse(parse_err) => {
+                        format!("failed to parse log in response: {}", parse_err)
+                    }
+                    _ => e.to_string(),
+                };
+                panic!("Failed to log in: {}", reason);
+            }
+        }
+    } else {
+        panic!("No previous session found and no credentials stored in config")
+    };
+
+    let filter = FilterDefinition::ignore_all().into();
+    let initial_sync_response = matrix_client
+        .send_request(assign!(sync_events::Request::new(), {
+            filter: Some(&filter),
+        }))
+        .await?;
+    let user_id = &config.username;
+    let not_senders = &[user_id.clone()];
+    let filter = {
+        let mut filter = FilterDefinition::empty();
+        filter.room.timeline.not_senders = not_senders;
+        filter
+    }
+    .into();
```

Comment:
I wasn't really aware of this possibility! Interesting, I wonder whether it would also work with the matrix-sdk or break sth. there.

Reply:
Yeah, it would be neat if it worked! :) I can't confirm whether it does since I'm using Conduit for testing, and filters haven't been implemented in Conduit yet.